### PR TITLE
Clean UriShouldNotBeHardcodedBase

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
I wanted to add a repro and while checking the code I've noticed some small things to improve on the code.

While I've changed my mind regarding the reproducer, I would still keep the code changes.

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs [hunk 1]
```diff
@@ -129,22 +127,22 @@ private bool IsInCheckedContext(TExpressionSyntax expression, SemanticModel mode
...
-                return methodSymbol != null &&
-                    argumentIndex.Value < methodSymbol.Parameters.Length
-                    && methodSymbol.Parameters[argumentIndex.Value].Name.SplitCamelCaseToWords()
-                        .Any(name => CheckedVariableNames.Contains(name));
+                return methodSymbol != null
+                       && argumentIndex.Value < methodSymbol.Parameters.Length
+                       && methodSymbol.Parameters[argumentIndex.Value]
+                                      .Name
+                                      .SplitCamelCaseToWords()
+                                      .Any(name => CheckedVariableNames.Contains(name));
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:145]: I'd be fine with having these on the single line.

Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:145]: I was about to ask for `x =>` and then realized this is also an option. Would it work?
```suggestion
                       && methodSymbol.Parameters[argumentIndex.Value]
                                      .Name
                                      .SplitCamelCaseToWords()
                                      .Any(CheckedVariableNames.Contains);
```

File: analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs [hunk 2]
```diff
@@ -91,8 +91,8 @@ protected override void Initialize(SonarAnalysisContext context)
                 c =>
                 {
                     var stringLiteral = (TLiteralExpressionSyntax)c.Node;
-                    if (UriRegex.IsMatch(GetLiteralText(stringLiteral)) &&
-                        IsInCheckedContext(stringLiteral, c.SemanticModel))
+                    if (UriRegex.IsMatch(GetLiteralText(stringLiteral))
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:94]: I would merge the nongeneric and generic base classes. The nongeneric doesn't seem to have practical value

**Review by pavel-mikula-sonarsource — APPROVED** [review]
LGTM

This class (due to its inheritors) desperately needs to inherit from `SonarDiagnosticAnalyzer<TSyntaxKind>` to remove bunch of noise - not in this PR

File: analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply analyzers/src/SonarAnalyzer.Common/Rules/UriShouldNotBeHardcodedBase.cs:94]: I will let this for another time.

**Pull request merged**